test/unit/fact_parser_test.rb

```diff
@@ -297,6 +297,22 @@ class FactParserTest < ActiveSupport::TestCase
       assert_equal '00:00:00:00:00:12', found.last[:macaddress]
     end
 
+    test "#suggested_primary_interface detects primary interface using DNS IPv6" do
+      parser.stubs(:interfaces).returns({
+        'br0' => {'ipaddress6' => 'fe80::10', 'macaddress' => '00:00:00:00:00:10'},
+        'em1' => {'ipaddress6' => 'fe80::20', 'macaddress' => '00:00:00:00:00:20'},
+        'em2' => {'ipaddress6' => 'fe80::30', 'macaddress' => '00:00:00:00:00:30'},
+        'bond0' => {'ipaddress6' => 'fe80::40', 'macaddress' => '00:00:00:00:00:40'},
```

Comment:
I wouldn't use `fe80` addresses. Link local addresses are special because you really need to include the interface as well. In https://community.theforeman.org/t/using-ips-and-domains-in-tests-documentation/16600 I've listed the values that are recommended for tests/documentation. For IPv6 in particular you should use `2001:db8::/32` per [RFC3849](https://tools.ietf.org/html/rfc3849).